src/NAR/Constructs/AddConstruct.java

```diff
@@ -0,0 +1,140 @@
+package NAR.Constructs;
+
+import NAR.Editor;
+import NAR.HighLevelFunc;
+import NAR.Parser;
+
+public class AddConstruct extends HighLevelFunc {
+    private double op1, op2, result;
+
+    public AddConstruct() {
+        super("add");
+    }
+
+    @Override
+    public boolean isCorrectSyntax(String statement) {
+        String lowerCase = statement.toLowerCase();
+        if (lowerCase.matches("^add (-?\\d+|\\w+|\\d+.\\d+) to (-?\\d+|\\w+|\\d+.\\d+)$")) {
+            return true;
+            //check syntax of the construct to execute when the condition is true
+        }
+        //^\s+add\s+\d+\s+to\s+\d+\s*$
+        //^add \d+ to \d+$
+        return false;
+    }
+
+    @Override
+    public void setArgs(String statement) {
+        String[] components = splitIntoComponents(statement.toLowerCase());
+        //TODO make sure components are not variables or parseInt will crash.
+        //Double[] vars=getVariables(components[0].substring(1),components[1].substring(1));
+        String v1,v2;
+        try {
+            if (components[0].charAt(0) == '-') {
+                v1 = components[0].substring(1);
+            } else {
+                v1 = components[0];
+            }
+            if (components[1].charAt(0) == '-') {
+                v2 = components[1].substring(1);
+            } else {
+                v2 = components[1];
+            }
+            Double[] vars = getVariables(v1, v2);
+
+            if (vars[0] != null) {
+                op1 = vars[0];
+                if (components[0].charAt(0) == '-') {
+                    op1 = -1 * op1;
+                }
+            } else {
+                if (components[0].charAt(0) == '-') {
```

Comment:
you don't need to check for the minus since the Double.parseDouble function will handle it for you